cloudahoywriter.lua

```diff
@@ -81,13 +111,19 @@ local recOnG = 0.2
 local recOnB = 0.2
 local recOnA = 0.8
 
-local maybe_write_data -- forward declaration
+-- forward declarations
+local maybe_write_data 
+local automatic_recording_state_check
 
 -- Runs on every draw. May write to file. May show UI.
 function CAWR_on_every_draw()
+    automatic_recording_state_check()
     maybe_write_data()
```

Comment:
Not related to this change - does flywithlua not give us a "flight loop callback"? It would be nice to isolate the data write and recording state check from the UI drawing loop, there are probably much fewer flight callbacks than UI draw callbacks.

Reply:
Good suggestion. See 2.2.37 - 2.2.46 in the FWL Manual for all the callbacks available. I think the closest thing to a "flight loop callback" is `do_every_frame()`. I re-read the manual again and it looks like I really should be doing the state checking in `do_every_frame()` and only the drawing in `do_every_draw()`.

More background info:
- All the samples I looked at were only using `do_every_draw()`, so I did that too.
- The docs for `do_every_draw()` say "the automatic DataRefs to variables transfer is disabled during a draw callback. So do not read or write DataRefs, use it only to draw your messages." 
  - But clearly that's not how it works since I am reading datarefs from variables in `do_every_draw()`.
  - I found an [old thread](https://forums.x-plane.org/index.php?/forums/topic/71120-dataref-confusion/&do=findComment&comment=769581) where the FWL says that read-only access is available during do_every_draw(), despite what the docs say.
  - I just [asked about this in the FWL help forum](https://forums.x-plane.org/index.php?/forums/topic/211462-reading-datarefs-in-do_every_draw/)

I'll try a change to do the state checking in `do_every_frame()` and only drawing in `do_every_draw()` anyway. That seems like better isolation of functionality.